samples/net/sockets/big_http_download/sample.yaml

```diff
@@ -16,3 +16,9 @@ tests:
     extra_configs:
       - CONFIG_NET_SOCKETS_POSIX_NAMES=n
       - CONFIG_POSIX_API=y
+  # CI-friendly config, using local host machine as
+  # a server and finite number of iterations.
+  sample.net.sockets.big_http_download.ci:
+    extra_configs:
+      - CONFIG_SAMPLE_BIG_HTTP_DL_URL="http://192.0.2.2:8980/static/vmlinuz"
```

Comment:
Better would be to put the values to an overlay and just use that if you need this kind of special configuration. If the values are here, then we run this for every sanitycheck run which is probably not needed.

Reply:
> If the values are here, then we run this for every sanitycheck run which is probably not needed.

That's true, but the whole idea is to use sanitycheck as the CI driver, i.e. sanitycheck is used to build binaries in the build system, and then produced binaries are submitted for testing on the hardware. I can propose following solutions:

* Mark sample.net.sockets.big_http_download.ci as a "slow" config (if that's possible for just one config of a sample.yaml).
* Vice-versa, mark the original sample.net.sockets.big_http_download as "slow" (which would be true, considering it ~ infinite runtime).
* Remove the original sample.net.sockets.big_http_download config, we won't lose anything, as new "sample.net.sockets.big_http_download.ci" is about the same re: code coverage (even slightly better, as runs code after the loop).
* Just replace the original sample.net.sockets.big_http_download to use those options, which kinda makes sense as sanitycheck output is intended for automated testing, and the new configuration.

Hopefully, at least one of these options would work.

Reply:
If you place these settings to the overlay, and use the overlay here like this:
```
extra_args: OVERLAY_CONFIG=overlay-big-http-dl.conf
```
Then it is possible for the sanitychecker to use it, and if needed user can manually compile the sample with this configuration.

Reply:
> If you place these settings to the overlay, and use the overlay here like this:
extra_args: OVERLAY_CONFIG=overlay-big-http-dl.conf

Sounds good, made the change. I named the overlay `overlay-ci.conf` though (and moved comment from sample.yaml to the .conf).